### PR TITLE
allow for wait periods between surveys

### PR DESCRIPTION
**Pull request by liyiy**
Do not show surveys if a wait period exists

related PR: https://github.com/PostHog/posthog/pull/17064

### CODE REVIEW

File: site.ts [hunk 1]
```diff
@@ -409,6 +409,17 @@ export function inject({ config, posthog }) {
         const nonAPISurveys = surveys.filter(survey => survey.type !== 'api')
         nonAPISurveys.forEach((survey) => {
             if (document.querySelectorAll("div[class^='PostHogSurvey']").length === 0) {
+                const surveyWaitPeriodInDays = survey.conditions?.seenSurveyWaitPeriodInDays
```
Comment (neilkakkar) [inline site.ts:412]: this could really use some tests 😅

**Review by neilkakkar — DISMISSED** [review]

**Review by neilkakkar — COMMENTED** [review]
Actually, having this condition on a survey doesn't make sense to me. What does it mean for one survey to have a wait period of 5 days, and another one to have a wait period of 2 days?

This implies that the shortest wait period would always come first, and if there are enough of these, the longer wait period one probably never shows up, because the timer resets after a survey is shown

**Comment by neilkakkar** [comment]
Ah, saw the other PR. This makes more sense to me as a global setting (i.e. for all surveys, don't show a new survey unless wait period has passed), rather than per survey, since in the latter case, you always have to set it otherwise all new surveys will bombard users.

I can see this make sense as an override, if, say, you have an important survey that trumps others, in which case this shows up even if the default wait period hasn't passed. But not sure if we really need it.

How are you envisioning this?

**Comment by liyiy** [comment]
> Ah, saw the other PR. This makes more sense to me as a global setting (i.e. for all surveys, don't show a new survey unless wait period has passed), rather than per survey, since in the latter case, you always have to set it otherwise all new surveys will bombard users.
> 
> I can see this make sense as an override, if, say, you have an important survey that trumps others, in which case this shows up even if the default wait period hasn't passed. But not sure if we really need it.
> 
> How are you envisioning this?

Hmmm, I think users will want a global setting too but I'm not sure they _don't_ also want a more specific per survey option. To be honest I think this could be a good place to start to see how people are using it and then extract into a global field later on, similar to appearance customizations for now.

Architecturally, we would just be adding a new global saved settings somewhere, and then allowing that to take precedence, I don't think anything we have here detracts from that future work 😮 

> you always have to set it otherwise all new surveys will bombard users

I would like to look at the stats on this a bit more 😅 like maybe users aren't actually facing this problem too much? Or even if they are, they don't know the problem, and maybe a "how to best" surveys guide in this case would be more helpful?

**Comment by neilkakkar** [comment]
Okay, sounds good 👍

File: site.ts (hunk 1 above)
Comment (liyiy) [reply site.ts:412]: imports are being kind of annoying for tests, I'll add this in a separate PR to avoid breaking everything 😅

**Pull request merged**